### PR TITLE
Add: CI for MSVC.

### PR DESCRIPTION
**Pull request by bjones1**
Add CI coverage for the MSVC compiler.

### CODE REVIEW

**Comment by bjones1** [comment]
Here's CI for MSVC, per the request in #184. Currently, it's failing until #184 is merged. There will probably be a few more bugs to fix after that, as well.

**Comment by bjones1** [comment]
I merged in the latest from main, then added fixes required to get CI passing with MSVC. Success!

**Comment by jbaldwin** [comment]
Nice! I'll get this one merged.  Awesome work.

File: CMakeLists.txt [hunk 1]
```diff
@@ -114,6 +115,10 @@ add_library(${PROJECT_NAME} STATIC ${LIBCORO_SOURCE_FILES})
 set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX PREFIX "")
 target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_20)
 target_include_directories(${PROJECT_NAME} PUBLIC include)
+if(MSVC)
```
Comment (jbaldwin) [inline CMakeLists.txt:118]: I'm curious too, but it's an easy enough workaround for now.

**Review by jbaldwin — APPROVED** [review]

**Pull request merged**